keps/sig-instrumentation/2305-metrics-cardinality-enforcement/README.md

```diff
@@ -53,9 +84,7 @@ The simple solution to this problem would be for each metric added to keep the u
 
 Instead, the proposed solution is we will be able to *dynamically configure* an allowlist of label values for a metric. By *dynamically configure*, we mean configure an allowlist *at runtime* rather than during the build/compile step. And by *at runtime*, we mean, more specifically, during the boot sequence for a Kubernetes component (and we mean daemons here, not CLI tools like kubectl).
 
-Brief aside: a Kubernetes component (which is a daemon) is an executable, which can be launched from the command line manually if desired. Components take a number of start-up configuration flags, which are passed into the component to modify execution paths (if curious, check out the [large amount of flags we have on the kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)). It is also possible to read configuration data from files (like yaml format) during the component boot sequence. This KEP does not have an opinion on the specific mechanism used to load config data into a Kubernetes binary during the boot sequence. What we *actually* care about, is just the fact that it is possible.
-
-Our design is thus config-ingestion agnostic.
+Brief aside: a Kubernetes component (which is a daemon) is an executable, which can be launched from the command line manually if desired. Components take a number of start-up configuration flags, which are passed into the component to modify execution paths (if curious, check out the [large amount of flags we have on the kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/)). Our design is going to add a flag that applies to all components to ingest the allowlist.
```

Comment:
how big can that allowlist get? enormous lists of CLI flags are really not convenient and of course there are limits - exec has a pretty high limit (1/4 of stack size in modern kernels, from what I read) but shells may have a much smaller buffer, like 2k.

Reply:
@logicalhan Do you have any insights about it? I was thinking if we can use the allowlist config file and pass in the path with CLI flag.

Reply:
I'm of the opinion that reading in from a file can be a beta feature. The most important thing here is that we provide the possibility of shunting off labels that have cardinality issues without necessarily having to sacrifice the entire metric.

Reply:
that's fine. So, for PRR we need:

 - add feature gate
 - note that in beta reading a file will be used instead (in addition?) to CLI flags
 -

Reply:
updated the two points. Thanks